packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -101,41 +101,54 @@ if (__DEV__) {
   } = require('./ReactDebugFiberPerf');
 
   var didWarnAboutStateTransition = false;
+  var didWarnSetStateChildContext = false;
+  var didWarnStateUpdateForUnmountedComponent = {};
 
   var warnAboutUpdateOnUnmounted = function(
     instance: React$ComponentType<any>,
```

Comment:
Let's change this to take `fiber` as argument instead. Then we can use `getComponentName(fiber) || 'ReactClass'` instead of duplicating its logic.

Reply:
I tried this and the tests were still failing. In the test output, getComponentName(fiber) was returning null (not sure why) and so instead of the expected component name, ReactClass was used as default value for component name and caused the test to fail. Not sure if I am missing something here.

Reply:
Which tests were failing? I tried this on your branch and it worked:

```diff
-  var warnAboutUpdateOnUnmounted = function(
-    instance: React$ComponentType<any>,
-  ) {
-    const ctor: Object = instance.constructor;
-    const componentName =
-      (ctor && (ctor.displayName || ctor.name)) || 'ReactClass';
+  var warnAboutUpdateOnUnmounted = function(fiber) {
+    const componentName = getComponentName(fiber) || 'ReactClass';
     if (didWarnStateUpdateForUnmountedComponent[componentName]) {
       return;
     }
@@ -1242,7 +1238,7 @@ module.exports = function<T, P, I, TI, PI, C, CC, CX, PL>(
         } else {
           if (__DEV__) {
             if (!isErrorRecovery && fiber.tag === ClassComponent) {
-              warnAboutUpdateOnUnmounted(fiber.stateNode);
+              warnAboutUpdateOnUnmounted(fiber);
             }
           }
           return;
```

Reply:
ooohh god.. yes i forgot to change it there, my bad. Sorry :(